### PR TITLE
adding flake8

### PR DESCRIPTION
**Pull request by RonTuretzky**
added flake8 yml and workflows directory 
#5

### CODE REVIEW

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      container: fedora:34
+      steps:
+        - name: Install Pipenv and Git
+          run: dnf install -y pipenv git
+        - name: Checkout code
+          uses: actions/checkout@v2
+        - name: Setup environment
+          run: pipenv sync --dev
+        - name: Run Flake8
+          run: pipenv run flake8 --max-line-length 120
```
Comment (NicolePauzner) [inline .github/workflows/flake8.yml:16]: Add new line at the end of the file

**Review by NicolePauzner — CHANGES_REQUESTED** [review]
Git commit message should start with a capital letter and be in imperative mood and not in indicative mood i.e. change adding to Add

File: .github/workflows/flake8.yml (hunk 1 above)
Comment (RonTuretzky) [reply .github/workflows/flake8.yml]: Done

**Comment by RonTuretzky** [comment]
> 

Amended

**Review by MosheSheena — APPROVED** [review]

File: .github/workflows/flake8.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+    flake8:
+      name: Check code with Flake8
+      runs-on: ubuntu-20.04
+      container: fedora:34
+      steps:
+        - name: Install Pipenv and Git
+          run: dnf install -y pipenv git
+        - name: Checkout code
+          uses: actions/checkout@v2
```
Comment (bennypowers) [inline .github/workflows/flake8.yml:12]:
```suggestion
          uses: actions/checkout@v3
```

File: .github/workflows/flake8.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        - name: Setup environment
+          run: pipenv sync --dev
+        - name: Run Flake8
+          run: pipenv run flake8 --max-line-length 120
```
Comment (bennypowers) [inline .github/workflows/flake8.yml:16]: probably better to use a [config file](https://flake8.pycqa.org/en/latest/user/configuration.html) but this is fine for now

File: .github/workflows/flake8.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: Flake8
+on: pull_request
+jobs:
+    flake8:
+      name: Check code with Flake8
+      runs-on: ubuntu-20.04
+      container: fedora:34
```
Comment (bennypowers) [inline .github/workflows/flake8.yml:7]: do you have a specific reason to use 34 and not 36?

**Review by bennypowers — CHANGES_REQUESTED** [review]

**Review by EliasAndry — DISMISSED** [review]

**Review by NicolePauzner — DISMISSED** [review]

**Review by FaezZangariya — APPROVED** [review]

File: .github/workflows/flake8.yml (hunk 3 above)
Comment (RonTuretzky) [reply .github/workflows/flake8.yml:16]: Adding it as an issue, will work on it in the future 👍

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (RonTuretzky) [reply .github/workflows/flake8.yml:7]: This is the version that was on HW1, I assume everyone is working off that. If there's not problem upgrading and it's not going to break anything backward then np yea lets upgrade.

**Review by EliasAndry — APPROVED** [review]

**Review by FaezZangariya — APPROVED** [review]

**Comment by kobihk** [comment]
@RonTuretzky please add the pipfiles otherwise we don't have flake8 to work with it

**Review by kobihk — APPROVED** [review]

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (kobihk) [reply .github/workflows/flake8.yml:7]: After we will make sure the newer version is working without any issues we can change it

**Pull request merged**